examples/axes.cu

```diff
@@ -171,3 +171,35 @@ using ctcs_types = nvbench::type_list<nvbench::int8_t,
                                       nvbench::float64_t>;
 NVBENCH_BENCH_TYPES(copy_type_conversion_sweep, NVBENCH_TYPE_AXES(ctcs_types, ctcs_types))
   .set_type_axes_names({"In", "Out"});
+
+// ==================================================================================
+// Passing `type_list` of typenames, and `enum_type_list` to build cartesian product
+// of typenames and integral constants
+
+template <typename ValueT, uint BLOCK_DIM>
+void copy_type_and_block_size_sweep(nvbench::state &state,
+                                    nvbench::type_list<ValueT, nvbench::enum_type<BLOCK_DIM>>)
```

Comment:
Remark: I think nvbench needs a different name for `enum_type`. The correct name is non-type template parameter, so maybe: `nvbench::nttp_list`, or just `nvbench::value_list`. You may also take inspiration here: http://wg21.link/P2781

Reply:
It's a good point -- I wouldn't get rid of the `enum_type`, since it's designed to be used with the [enum type string](https://github.com/NVIDIA/nvbench/blob/main/examples/enums.cu#L106-L166) machinery, and the name is more discoverable for the common case of using named enums (eg tuning params in CUB).

Rather than introduce a new `nttp` type, let's just just update this example to use `type_list<integral_constant>` directly for now, rather than using `enum_type` for non-enum values.

@oleksandr-pavlyk could you make that change?